### PR TITLE
Improve typing

### PR DESCRIPTION
**Pull request by mrkkrp**
So here are some changes that I think are good. @tzemanovic, please take a look.

### CODE REVIEW

**Comment by tzemanovic** [comment]
Looks great, thanks!

File: Web/Forma.hs [hunk 1]
```diff
@@ -88,7 +87,6 @@ where
 import Control.Applicative
```
Comment (tzemanovic) [inline Web/Forma.hs]: Just one detail, can you export `FieldError(..)`?

**Comment by mrkkrp** [comment]
That would allow creation of `FieldError`s with no errors inside, which I'd like to forbid. What do you need the data constructor for?

**Comment by tzemanovic** [comment]
Good point, how about a fn `unFieldError`? I'd like to format the errors myself a bit differently

**Comment by tzemanovic** [comment]
it might be nice to return `Map [SelectedName names] Value` or even `[([SelectedName names], Value)]` to make it easier to work with

**Comment by mrkkrp** [comment]
I'll add `unFieldError`. Maybe we just should change how errors for nested fields are formatted, though.

**Comment by tzemanovic** [comment]
If you prefer this format too, I can make a PR for it

**Comment by mrkkrp** [comment]
I don't have strong preferences. Whatever is easier for frontend people to work with.

**Comment by tzemanovic** [comment]
I guess it's quite trivial to convert it, but it seems like a nicer default because it follows the format of the input.

**Comment by mrkkrp** [comment]
But then there will be an inconsistency between format of validation errors and parse error (we provide `field' which points where parsing error happened).

**Comment by tzemanovic** [comment]
I need to format both parsing and validation errors like that

**Comment by mrkkrp** [comment]
Give it a go then, I think it's OK to have this as default.

**Pull request merged**